src/Plugin/Field/FieldFormatter/StrawberryBaseFormatter.php

```diff
@@ -0,0 +1,246 @@
+      $configuration['third_party_settings'],
+      $container->get('http_client'),
+      $container->get('config.factory')
+    );
+  }
+
+  /**
+   * @param string $plugin_id
+   * @param array $settings
+   * @param mixed $plugin_definition
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   * @param GuzzleHttp\ClientInterface $httpClient
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The definition of the field to which the formatter is associated.
+   * @param string $label
+   *   The formatter settings.
+   * @param $view_mode
+   *   The view mode.
+   * @param array $third_party_settings
+   *   Any third party settings.
+   */
+  public function __construct(
+    $plugin_id,
+    $plugin_definition,
+    FieldDefinitionInterface $field_definition,
+    array $settings,
+    $label,
+    $view_mode,
+    array $third_party_settings,
+    ClientInterface $httpClient,
+    ConfigFactoryInterface $config_factory
+  ) {
+    parent::__construct(
+      $plugin_id,
+      $plugin_definition,
+      $field_definition,
+      $settings,
+      $label,
+      $view_mode,
+      $third_party_settings
+    );
+    $this->httpClient = $httpClient;
+    $config = $config_factory->get('format_strawberryfield.iiif_settings');
+    $this->iiif_pub_server = $config->get('pub_server_url');
+    $this->iiif_int_server = $config->get('int_server_url');
+  }
+
+  //todo @marlo -- use config factory, Docs say best practice is to use dep injection, not \Drupal::
+  public static function defaultSettings() {
+    return [
+      'iiif_base_url' => \Drupal::config('format_strawberryfield.iiif_settings')->getOriginal('pub_server_url', FALSE),
+      'iiif_base_url_internal' => \Drupal::config('format_strawberryfield.iiif_settings')->getOriginal('int_server_url', FALSE),
+      ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function IiifSettingsForm(array $form, FormStateInterface $form_state) {
+    return [
+      'iiif_base_url' => [
+        '#type' => 'url',
+        '#title' => $this->t('Base Public accesible URL of your IIIF Media Server'),
+        '#default_value' => $this->getSetting('iiif_base_url'),
+        '#required' => TRUE
+      ],
+      'iiif_base_url_internal' => [
+        '#type' => 'url',
+        '#title' => $this->t('Base URL of your IIIF Media Server accesible from inside this Webserver'),
+        '#default_value' => $this->getSetting('iiif_base_url_internal'),
+        '#required' => TRUE,
+        '#element_validate' => [[$this, 'validateIiifUrl']],
+      ],
+      'iiif_reset_button' => [
+        '#type' => 'button',
+        '#name' => 'iiif_reset_button',
+        '#value' => t('Reset to IIIF Defaults'),
+        '#executes_submit_callback' => TRUE,
+        '#submit' => [[$this, 'resetIiifDefaults']],
+      ],
+    ];
+  }
+
+
+  public function validateIiifUrl(array $form, FormStateInterface $form_state) {
```

Comment:
Wrong signature!
`&$element, FormStateInterface $form_state, &$complete_form`. The $element is passed by reference always. and you are missing also the  &$complete_form